packages/aws-cdk-lib/aws-codepipeline/lib/trigger.ts

```diff
@@ -145,10 +268,33 @@ export class Trigger {
         };
         return { tags };
       });
+      const pullRequest: CfnPipeline.GitPullRequestFilterProperty[] | undefined = pullRequestFilter?.map(filter => {
```

Comment:
This is getting a bit messy.
Can you please create the `private renderPushFilter(pushFilter?: GitPushFilter[]): CfnPipeline.GitPushFilterProperty[] | undefined` and the `private renderPullRequestFilter(pullRequest?: GitPullRequestFilter[]):  CfnPipeline.GitPullRequestFilterProperty[] | undefined` methods to handle rendering for specific filters?

Reply:
I changed to create the rendering methods.

https://github.com/aws/aws-cdk/pull/29128/commits/3f057a530dac52493d6c6348455bd8f37e443582